www_src/components/modal/modal.jsx

```diff
@@ -0,0 +1,50 @@
+var React = require('react');
+var dispatcher = require('../../lib/dispatcher');
+
+var Modal = React.createClass({
+  getInitialState: function () {
+    return {
+      isVisible: false
```

Comment:
we're also using `isVisible` as a special state property for determining whether views are "in view" or frozen off screen into cache, so it might be safer to use a different term?